chain/app/db_app.go

```diff
@@ -244,45 +250,27 @@ func (app *dbBandApp) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBl
 
 	events := res.GetEvents()
 	for _, event := range events {
-		if event.Type == zoracle.EventTypeRequestExecute {
-			var requestID int64
-			var resolveStatus zoracle.ResolveStatus
-			for _, kv := range event.Attributes {
-				if string(kv.Key) == zoracle.AttributeKeyRequestID {
-					requestID, err = strconv.ParseInt(string(kv.Value), 10, 64)
-					if err != nil {
-						panic(err)
-					}
-				} else if string(kv.Key) == zoracle.AttributeKeyResolveStatus {
-					numResolveStatus, err := strconv.ParseInt(string(kv.Value), 10, 8)
-					if err != nil {
-						panic(err)
-					}
-					resolveStatus = zoracle.ResolveStatus(numResolveStatus)
-				}
-			}
-			// Get result from keeper
-			var rawResult []byte
-			rawResult = nil
-			if resolveStatus == 1 {
-				id := zoracle.RequestID(requestID)
-				request, sdkErr := app.ZoracleKeeper.GetRequest(app.DeliverContext, id)
-				if sdkErr != nil {
-					panic(err)
-				}
-				result, sdkErr := app.ZoracleKeeper.GetResult(app.DeliverContext, id, request.OracleScriptID, request.Calldata)
-				if sdkErr != nil {
-					panic(err)
-				}
-				rawResult = result.Data
-			}
-			err := app.dbBand.ResolveRequest(requestID, resolveStatus, rawResult)
-			if err != nil {
-				panic(err)
-			}
-		}
+		app.dbBand.HandleEndblockEvent(event)
 	}
 
+	// Update bonding validator status
+	app.StakingKeeper.IterateValidators(app.DeliverContext,
```

Comment:
If we are looping like this, then why don't we use this loop to update other info as well? Tokens, Jails, DelegatorShares, etc can be updated here. and we will be very sure that we won't mess up. Would that be better?

Reply:
Now that I think about it. Let's just remove this status thing. It looks quite inefficient and I don't think we need to show it for now. We can certainly revisit later.